### PR TITLE
Introduce termial function to prefer more recent versions randomly

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

File: thoth/adviser/state.py [hunk 1]
```diff
@@ -170,12 +171,93 @@ def get_first_unresolved_dependency(self) -> Tuple[str, str, str]:
...
+    @classmethod
+    def _random_termial(cls, n: int) -> int:
+        """Compute a random number x such as 0 <= x < n, use termial function not to spread numbers uniformly.
+
+        To prefer lower numbers more often, termial function is used to assign "weights" for
+        numbers - then random uniform is used in conjuction with weights.
+
+        An illustrative example can be cls._random_terminal(4).
+
+          cls._terminal_function(4) == 10
```
Comment (pacospace) [inline thoth/adviser/state.py]: `_random_termial` and `_termial_function`?

Comment (fridex) [reply thoth/adviser/state.py]: `_random_termial` calls `_termial_function` - hence the comment. I will adjust the docstring.

Comment (pacospace) [reply thoth/adviser/state.py]: are there typos also in the docstring?

Comment (fridex) [reply thoth/adviser/state.py]: Agh, right :+1: It's termial, fixed in 354a9a7

Comment (pacospace) [reply thoth/adviser/state.py]: Then it's ready for me :) Thanks, sorry for being picky :P

Comment (fridex) [reply thoth/adviser/state.py]: Actually, I love picky reviews :smile_cat:

**Pull request merged**